test/Core.Test/OrganizationFeatures/OrganizationSponsorships/FamiliesForEnterprise/Cloud/CloudSyncSponsorshipsCommandTests.cs

```diff
@@ -88,5 +88,94 @@ await sutProvider.GetDependency<IOrganizationSponsorshipRepository>()
+            await sutProvider.GetDependency<IOrganizationSponsorshipRepository>()
+                .Received(1)
+                .UpsertManyAsync(Arg.Is<IEnumerable<OrganizationSponsorship>>(sponsorships => sponsorships.Count() == 2));
+
+            // Neither were marked as delete, should not have deleted
+            await sutProvider.GetDependency<IOrganizationSponsorshipRepository>()
+                .DidNotReceiveWithAnyArgs()
+                .DeleteManyAsync(default);
+
+            // Only one sponsorship was new so it should only send one
+            Assert.Single(toEmailSponsorships);
+        }
+
+        [Theory]
+        [BitAutoData]
+        public async Task SyncOrganization_TwoToDelete_OneCanDelete_Success(SutProvider<CloudSyncSponsorshipsCommand> sutProvider,
+            Organization sponsoringOrganization, OrganizationSponsorship canDeleteSponsorship, OrganizationSponsorship cannotDeleteSponsorship)
+        {
+            // Arrange
+            sponsoringOrganization.PlanType = PlanType.EnterpriseAnnually;
+
+            canDeleteSponsorship.ToDelete = true;
+            canDeleteSponsorship.SponsoredOrganizationId = null;
+
+            cannotDeleteSponsorship.ToDelete = true;
+            cannotDeleteSponsorship.SponsoredOrganizationId = Guid.NewGuid();
+
+            sutProvider.GetDependency<IOrganizationSponsorshipRepository>()
+                .GetManyBySponsoringOrganizationAsync(sponsoringOrganization.Id)
+                .Returns(new List<OrganizationSponsorship>
+                {
+                    canDeleteSponsorship,
+                    cannotDeleteSponsorship,
+                });
+
+            // Act
+            var (syncData, toEmailSponsorships) = await sutProvider.Sut.SyncOrganization(sponsoringOrganization, new[]
+            {
+                new OrganizationSponsorshipData(canDeleteSponsorship),
+                new OrganizationSponsorshipData(cannotDeleteSponsorship),
+            });
+
+            // Assert
+
+            await sutProvider.GetDependency<IOrganizationSponsorshipRepository>()
+                .Received(1)
+                .UpsertManyAsync(Arg.Is<IEnumerable<OrganizationSponsorship>>(sponsorships => sponsorships.Count() == 2));
+
+            // Deletes the sponsorship that had delete requested and is not sponsoring an org
+            await sutProvider.GetDependency<IOrganizationSponsorshipRepository>()
+                .Received(1)
+                .DeleteManyAsync(Arg.Is<IEnumerable<Guid>>(toDeleteIds =>
+                    toDeleteIds.Count() == 1 && toDeleteIds.ElementAt(0) == canDeleteSponsorship.Id));
+        }
+
+        [Theory]
+        [BitAutoData]
+        public async Task SyncOrganization_BadData_DoesNotSave(SutProvider<CloudSyncSponsorshipsCommand> sutProvider,
+            Organization sponsoringOrganization, OrganizationSponsorship badOrganizationSponsorship)
+        {
+            sponsoringOrganization.PlanType = PlanType.EnterpriseAnnually;
```

Comment:
Does this test do anything?

Reply:
It did not, I have added the test I meant to be there.